### PR TITLE
feat(server): allow named url shortening (#40)

### LINKED ISSUE #40: ﻿Only one URL at a time can be shortened when filenames aren't random
The URL is always stored as `url`, so any URL overwrites the previous. I propose accepting `curl -F url=http://example.com;filename=foo` as a fix consistent with how Rustypaste works with files.

### PR DESCRIPTION
**Pull request by Vaelatern**
## Description

I ran the filename header through the URL shortening route

## Motivation and Context

This lets me use rustypaste to link to a specific link using a specified short name (for ergonomics).

Also fixes #40 by the by.

## Changelog Entry

```
### Added

- Enable naming shortened URLs beyond using a random string or the text "url"
````
-->

## Types of Changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tessus** [comment]
Thanks for the PR. Can you also please add test cases and fixtures?

**Comment by tessus** [comment]
There are a few things to fix. The formatting checks, lints, and tests fail. These have to be fixed and new tests should be added that handle your use case.

**Review by tessus — CHANGES_REQUESTED** [review]
Unfortunately there are still issues with this PR.

May I suggest to go through the checklist?

- `cargo fmt --check --`
- `cargo clippy`
- `cargo test -- --test-threads 1`
- `./fixtures/test-fixtures.sh`

Currently the code does not even compile.

**Comment by Vaelatern** [comment]
I've pushed an update now, it passes CI on my own fork.

**Review by tessus — APPROVED** [review]
Nice, this looks good. Thanks.

**Comment by tessus** [comment]
@orhun can you please have a look? I promised I would only merge dependency PRs. ;-)

@Vaelatern let's wait for orhun's approval, since he's the owner of the project. I can't merge without his ok.

**Comment by Vaelatern** [comment]
Sounds good!

I just started using this today, noticed this hole, and wanted to contribute a fix.

**Comment by Vaelatern** [comment]
Apparently I need to sign the commits too

**Comment by tessus** [comment]
why? nope, you don't. This project does not have a requirement of signed commits.

**Comment by tessus** [comment]
oops, yeah. I am sorry. this policy must be new.

**Comment by Vaelatern** [comment]
Now pushed with signed commit, both this and #374  (which I'm about to refactor a tiny bit)

**Comment by tessus** [comment]
Thanks.

**Review by orhun — APPROVED** [review]
Good stuff 💯 

Thanks!

**Pull request merged**
